jscomp/ml/predef.ml

```diff
@@ -219,9 +221,19 @@ let common_initial_env add_type add_extension empty_env =
   and decl_dict =
     let tvar = newgenvar() in
     {decl_abstr with
+      type_attributes = [(Location.mknoloc "res.dict", Parsetree.PStr [])];
       type_params = [tvar];
       type_arity = 1;
-      type_variance = [Variance.full]}
+      type_variance = [Variance.full];
+      type_kind = Type_record ([{
+          ld_id = ident_anyOtherField;
+          ld_attributes = [(Location.mknoloc "res.optional", Parsetree.PStr []); (Location.mknoloc "res.dictMagicField", Parsetree.PStr [])];
```

Comment:
perhaps add high-level comment about what the trick is "creare a record type with a fields called xxx"
perhaps make the magic field more "**$!!!mag1x"